streams/src/test/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreatorTest.java

```diff
@@ -85,9 +89,20 @@
 
     @Test
     public void shouldCreateThreadProducerIfEosDisabled() {
+        final Map<String, Object> mockProducerConfig = mock(Map.class);
+        expect(mockProducerConfig.containsKey(ProducerConfig.TRANSACTIONAL_ID_CONFIG)).andReturn(false);
+        replay(mockProducerConfig);
+
+        config = new StreamsConfig(properties) {
+            @Override
+            public Map<String, Object> getProducerConfigs(final String clientId) {
+                return mockProducerConfig;
+            }
+        };
         createTasks();
 
         assertThat(mockClientSupplier.producers.size(), is(1));
+        verify(mockProducerConfig);
```

Comment:
Adding more checks for this and other tests. Here: verify that we don't call `put(...)` (ie, don't set a `transactional.id`) for the non-eos case

Reply:
Sounds great.